src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -0,0 +1,28 @@
+import { type Token } from '@dfinity/utils';
+
+/**
+ * Calculates the maximum amount for a transaction.
+ *
+ * @param {Object} params
+ * @param {bigint | undefined} params.balance The balance of the account in E8S.
+ * @param {bigint | undefined} params.fee The fee of the transaction in E8S.
+ * @param {bigint | undefined} params.maxAmount The maximum amount of the transaction not counting the fees.
+ * @returns {number} The maximum amount for the transaction.
+ */
+export const getMaxTransactionAmount = ({
+	balance = 0n,
+	fee = 0n,
+	maxAmount,
+	token
+}: {
+	balance?: bigint;
+	fee?: bigint;
+	maxAmount?: bigint;
```

Comment:
do we need this parameter `maxAmount` in Oisy's context?

Reply:
not for now, agreed, i will remove it; but it is just thinking about possible tokens with a maximum amount possible to be transferred, or specific transactions with a maximum possible amount